src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java

```diff
@@ -0,0 +1,73 @@
+package com.wikia.webdriver.testcases.mercurytests;
+
+import com.wikia.webdriver.common.contentpatterns.MercurySubpages;
+import com.wikia.webdriver.common.contentpatterns.MercuryWikis;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.InBrowser;
+import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.helpers.Emulator;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.common.Navigate;
+import com.wikia.webdriver.elements.mercury.components.Navigation;
+import com.wikia.webdriver.elements.mercury.components.TopBar;
+import org.testng.annotations.Test;
+
+@Execute(onWikia = MercuryWikis.MERCURY_AUTOMATION_TESTING)
+@InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+public class TopBarTests extends NewTestTemplate {
+
+  private TopBar topBar;
+  private Navigation navigation;
+
+  private void init() {
+    this.topBar = new TopBar(driver);
+    this.navigation = new Navigation(driver);
+
+    new Navigate(driver).toPage(MercurySubpages.MAIN_PAGE);
+  }
+
+  @Test(groups = "mercury_topbar_topBarIsAlwaysVisible")
+  public void mercury_topbar_topBarIsAlwaysVisible() {
+    init();
+
+    Assertion.assertTrue(topBar.isNavigationBarVisible());
+    Assertion.assertTrue(topBar.isLogoVisible());
+    Assertion.assertTrue(topBar.isHamburgerIconVisible());
+    Assertion.assertTrue(topBar.isSearchIconVisible());
+
+    driver.executeScript("window.scrollTo(100, document.body.scrollHeight)");
+    Assertion.assertTrue(topBar.isNavigationBarVisible());
+    Assertion.assertTrue(topBar.isLogoVisible());
+    Assertion.assertTrue(topBar.isHamburgerIconVisible());
+    Assertion.assertTrue(topBar.isSearchIconVisible());
+  }
+
+  @Test(groups = "mercury_topbar_closeButtonAppears")
+  public void mercury_topbar_closeButtonAppears() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)